diff --git a/build.linux64ARMv8/squeak.cog.spur/build/mvm b/build.linux64ARMv8/squeak.cog.spur/build/mvm
index d73f46d..ec8ce9e 100755
--- a/build.linux64ARMv8/squeak.cog.spur/build/mvm
+++ b/build.linux64ARMv8/squeak.cog.spur/build/mvm
@@ -4,7 +4,7 @@ set -e
 INSTALLDIR=sqcogspur64ARMv8linuxht
 # armv8.N-a all fail in signalSemaphoreWithIndex for N in 1,2,3,4,5
 MACHINE="-march=armv8-a -mtune=cortex-a72"
-OPT="-g -O3 -DNDEBUG -DDEBUGVM=0"
+OPT="-g -O3 -DNDEBUG -DDEBUGVM=0 -DMUSL -D_GNU_SOURCES"
 
 if [ $# -ge 1 ]; then
 	INSTALLDIR="$1"; shift
@@ -20,7 +20,7 @@ test -f plugins.int || (test -f ../plugins.int && cp -p ../plugins.int . || cp -
 test -f plugins.ext || (test -f ../plugins.ext && cp -p ../plugins.ext . || cp -p ../../plugins.ext .)
 test -f config.h || ../../../platforms/unix/config/configure \
 	--with-vmversion=5.0 --with-src=spur64src \
-	--without-vm-display-fbdev --without-npsqueak \
+	--with-vm-display-fbdev --without-npsqueak \
 	CFLAGS="$MACHINE $OPT -DCOGMTVM=0 -DDUAL_MAPPED_CODE_ZONE=1" \
 	LIBS="-lrt"
 
diff --git a/platforms/Cross/vm/sqVirtualMachine.c b/platforms/Cross/vm/sqVirtualMachine.c
index 2e9ae66..f61bcdd 100755
--- a/platforms/Cross/vm/sqVirtualMachine.c
+++ b/platforms/Cross/vm/sqVirtualMachine.c
@@ -1,4 +1,5 @@
 #include <math.h>
+#include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -581,9 +582,11 @@ struct VirtualMachine* sqGetInterpreterProxy(void)
 /* This lives here for now but belongs somewhere else.
  * platforms/Cross/vm/sqStuff.c??
  */
+#ifndef MUSL
 #define STDOUT_STACK_SZ 5
 static int stdoutStackIdx = -1;
 static FILE stdoutStack[STDOUT_STACK_SZ];
+#endif
 
 /* N.B. As of cygwin 1.5.25 fopen("crash.dmp","a") DOES NOT WORK!  crash.dmp
  * contains garbled output as if the file pointer gets set to the start of the
@@ -607,6 +610,7 @@ fopen_for_append(char *filename)
 # define fopen_for_append(filename) fopen(filename,"a+")
 #endif
 
+#ifndef MUSL
 void
 pushOutputFile(char *filenameOrStdioIndex)
 {
@@ -656,6 +660,7 @@ popOutputFile()
 	}
 	*stdout = stdoutStack[stdoutStackIdx--];
 }
+#endif
 
 void
 printPhaseTime(int phase)
diff --git a/platforms/unix/vm/sqUnixMain.c b/platforms/unix/vm/sqUnixMain.c
index 7156637..51d6db1 100644
--- a/platforms/unix/vm/sqUnixMain.c
+++ b/platforms/unix/vm/sqUnixMain.c
@@ -862,6 +862,11 @@ sqInt  primitivePluginRequestState(void)	{ return dpy->primitivePluginRequestSta
 
 /*** errors ***/
 
+#ifdef MUSL
+void pushOutputFile(char *fileNameOrStdioIndex) {;}
+void popOutputFile() {;}
+#endif
+
 static void outOfMemory(void)
 {
   /* pushing stderr outputs the error report on stderr instead of stdout */
